### PR TITLE
CASSGO-7 Change Batch API to be consistent with Query()

### PR DESCRIPTION
**Pull request by tengu-alt**
According to issue #1187, I have refactored the Query() method for the batch and implemented the Exec() method for the batch structure to save the general pattern of the query execution, and saved the older method (session.ExecuteBatch()) for backward compatibility. 
This means the batch for now behaves the same way as query.

### CODE REVIEW

**Review by testisnullus — APPROVED** [review]

**Comment by joao-r-reis** [comment]
@tengu-alt we should also add a new `Batch()` function and deprecate the existing `NewBatch` (basically a "soft" rename of `NewBatch()` -> `Batch()` without breaking users) and deprecate `Session.ExecuteBatch`. 

`NewBatch` and `ExecuteBatch` can then be removed in 3.0 so that users have more than enough time to transition to the new names.

**Comment by tengu-alt** [comment]
> @tengu-alt we should also add a new `Batch()` function and deprecate the existing `NewBatch` (basically a "soft" rename of `NewBatch()` -> `Batch()` without breaking users) and deprecate `Session.ExecuteBatch`.
> 
> `NewBatch` and `ExecuteBatch` can then be removed in 3.0 so that users have more than enough time to transition to the new names.

@joao-r-reis I do not clearly understand. The `NewBatch()` function is already marked as `deprecated` and `session.NewBatch`  is recommended to use instead. Also I can't name the function same as struct (Batch) because the `Batch` is already declared.

**Comment by joao-r-reis** [comment]
`Session.NewBatch()` should be deprecated now, global `NewBatch()` can be removed since it was deprecated already and `Session.Batch()` should be added to replace `Session.NewBatch()` (keep the same functionality). 

`Session.Batch()` does not have a conflict with the type `Batch`, you can have both (in fact this already exists with `Query()`)

The original issue mentions the rename `Session.NewBatch()` to `Session.Batch()` already, my only additional suggestion here is to keep and deprecate `Session.NewBatch()` for now AND add the new `Session.Batch()`.

**Comment by joao-r-reis** [comment]
global `NewBatch()` removal is already handled by https://github.com/apache/cassandra-gocql-driver/pull/1762 so it doesn't have to happen here

**Comment by tengu-alt** [comment]
> `Session.NewBatch()` should be deprecated now, global `NewBatch()` can be removed since it was deprecated already and `Session.Batch()` should be added to replace `Session.NewBatch()` (keep the same functionality).
> 
> `Session.Batch()` does not have a conflict with the type `Batch`, you can have both (in fact this already exists with `Query()`)
> 
> The original issue mentions the rename `Session.NewBatch()` to `Session.Batch()` already, my only additional suggestion here is to keep and deprecate `Session.NewBatch()` for now AND add the new `Session.Batch()`.

I see, thank you. Done, also I've changed the `Session.NewBatch()` usage in .doc and tests.

File: session.go
Comment (worryg0d) [inline session.go:739]: I think you should add a doc here because it is meant to be used by users. Maybe use the same description as for `Session.ExecuteBatch()`.

Also, add a doc for `Session.Batch()` too.

Comment (tengu-alt) [reply session.go:739]: Agree, done.

File: session.go
Comment (joao-r-reis) [inline session.go]: duplicate code (`NewBatch`)

**Review by joao-r-reis — COMMENTED** [review]
LGTM, just 1 comment about the duplicate code in the new function

File: session.go
Comment (tengu-alt) [reply session.go]: I kept the `session.NewBatch` and marked it as deprecated according to your suggestion.

Comment (joao-r-reis) [reply session.go]: Yes thanks but please create a private function that both can call or make one of them call the other so we don't have the duplicate code.

Comment (tengu-alt) [reply session.go]: Got it! Fixed.

**Review by joao-r-reis — APPROVED** [review]

**Comment by joao-r-reis** [comment]
Please format your commit message following the rules stated in [CONTRIBUTING.md](https://github.com/apache/cassandra-gocql-driver/blob/trunk/CONTRIBUTING.md#commit-message) and add this JIRA to the `CHANGELOG.md` in the `Changed` section of `Unreleased`, thanks!

We need another committer +1 to merge this.

**Comment by joao-r-reis** [comment]
Sorry I merged another PR which is causing a git conflict on `CHANGELOG.md` can you fix it and I'll merge this right after? Otherwise I can merge manually and fix it in place

**Comment by tengu-alt** [comment]
> Sorry I merged another PR which is causing a git conflict on `CHANGELOG.md` can you fix it and I'll merge this right after? Otherwise I can merge manually and fix it in place

Fixed! Thank you for the hint.

**Comment by joao-r-reis** [comment]
@tengu-alt we forgot to deprecate session.ExecuteBatch(), do you have some free time to work on this? I created https://issues.apache.org/jira/browse/CASSGO-57 for it

**Comment by joao-r-reis** [comment]
Oh and there's a few other session methods to execute batches that should also be moved to the Batch type

**Comment by tengu-alt** [comment]
> @tengu-alt we forgot to deprecate session.ExecuteBatch()

Got It I'll take it to work.

> Oh and there's a few other session methods to execute batches that should also be moved to the Batch type

I'll mark them as deprecated and create similar methods for batch. (If I understand correctly - these methods: `ExecuteBatchCAS`, `MapExecuteBatchCAS` )

**Pull request merged**
